### PR TITLE
Feature/context decorator

### PR DESCRIPTION
**Pull request by NicolasBissig**

### CODE REVIEW

File: test/decorators/context.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    page: number;
+    context: AzureContext;
+};
+
+class ContextEcho {
+    @HttpFunction()
+    static async httpTrigger(
+        @PathParameter('page') page: number,
+        @Context() context: AzureContext
+    ): Promise<ContextEchoResponse> {
```
Comment (NicolasBissig) [inline test/decorators/context.test.ts:14]: It is now possible to specifically pass the Context argument, always provided as the first argument by azure, to a special positined argument.

File: test/decorators/context.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { Context as AzureContext, HttpRequest } from '@azure/functions';
```
Comment (NicolasBissig) [inline test/decorators/context.test.ts:1]: Due to the decorator and Azure Context having the same name, this has to be done... ugly. Any ideas?

File: src/decorators/context.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    args: any[]
+) {
+    let contextParameter: number[] = Reflect.getOwnMetadata(ContextMetaDataKey, target, propertyName);
+    if (contextParameter) {
+        if (contextParameter.length !== 1) {
+            throw new Error('only one @Context parameter is allowed');
+        }
+
+        const paramIndex = contextParameter[0];
+        args[paramIndex] = context;
```
Comment (NicolasBissig) [inline src/decorators/context.ts:29]: In this part, we assume that the first argument from azure is always the context. Here some checks could be added if it actually is a context. But the current context type guard is specific to HTTP functions...

File: test/decorators/context.test.ts (hunk 2 above)
Comment (sladkoff) [reply test/decorators/context.test.ts:1]: No, this is completely fine IMO.

**Comment by sladkoff** [comment]
@NicolasBissig LGTM. This is still marked as a draft? Feel free to merge.

**Review by sladkoff — APPROVED** [review]

**Pull request merged**